### DISCUSSION TITLE
How can I correctly run one single test from a file?

### DISCUSSION THREAD

**Discussion by riskoviv** [orta/vscode-jest#876] [Q&A] [answered]
I write tests in TypeScript. When I try to run one single test or describe block by clicking on ![image](https://user-images.githubusercontent.com/44531490/172637196-f87a8948-f3fb-4604-a728-bb6f4aac35e0.png) icon or right-click -> "Run test" on it, I've got this error in terminal:
![image](https://user-images.githubusercontent.com/44531490/172640331-3ef912f8-60b9-46c0-a08e-95a2ed76d966.png)
`testMatch`, `testPathIgnorePatterns` and `testRegex` are have default jest config values, no changes made.

This error occurs regardless of jest.autoRun extension settings.
However running tests on saving a file works fine
I'm using these settings:
```
"jest.autoRun": {
   "watch": false,
   "onSave": "test-file"
}
```
and running tests by `npm test` works too.

I just want to be able to run tests individually, not only the whole test files. Do I need to change some settings or something else to make it work?

_P.S.: I know that `test.only` exists, but I want to use the same feature provided by this plugin since it exists._

**Comment by connectdotz** [comment]
does your npm happen to be >= 7.0? If true, then this is a known issue, see #838, #852
You can find a few workarounds suggested [here](https://github.com/jest-community/vscode-jest/issues/838#issuecomment-1051323196). Let me know if it works for you.

**Reply by riskoviv** [reply]
Okay, got it, my npm is 8.5.0. I think then I'll better just keep using test.only and describe.only for now (pretty useful feature btw). I see that the most appropriate solution for the issue is switching to yarn, but I can't do it right now.

**Comment by connectdotz** [comment, marked as answer]
@riskoviv added another workaround that might help - directly invoking the command to by-pass `npm run`, see the 3rd suggestion in the [comment](https://github.com/jest-community/vscode-jest/issues/838#issuecomment-1051323196)

**Reply by riskoviv** [reply]
I think this option value assumes using React, but I don't have `react-scripts.cmd` mentioned there because I'm not using React currently (and to be honest right now I'm working on educational project using just TypeScript w/o libraries).
But instead of it I set **jestCommandLine** option to `node_modules\\.bin\\jest.cmd` and now running single tests works perfectly. Thank you!

**Reply by connectdotz** [reply]
yup, you got the idea 👍 . Basically, just directly invoke whatever your package.json test script is doing with the `jest.jestCommandLine` setting to by-pass `npm run`.

**Reply by riskoviv** [reply]
By the way! If `jest.jestCommandLine` option value is empty (as it is by default), everything works correctly too. It looks like it didn't work correctly only if it had value `npm test --` which I did set then (I forgot to mention this in the first message)
